tests/phpunit/CRM/Civicase/Service/CaseCategoryInstanceTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+use CRM_Civicase_Service_CaseCategoryInstance as CaseCategoryInstanceService;
+use CRM_Civicase_Test_Fabricator_CaseCategoryInstance as CaseCategoryInstanceFabricator;
+use CRM_Civicase_Test_Fabricator_CaseCategory as CaseCategoryFabricator;
+
+/**
+ * Runs tests on CaseCategoryInstance Service tests.
+ *
+ * @group headless
+ */
+class CRM_Civicase_Service_CaseCategoryInstanceTest extends BaseHeadlessTest {
+
+  /**
+   * Test getCaseCategoryInstances without parameters.
+   *
+   * Verify that all category instances are returned, and the result is
+   * formatted as expected.
+   */
+  public function testGetCategoryInstances() {
+    // Generate random category instances.
+    $categoryInstanceOne = CaseCategoryInstanceFabricator::fabricate();
+    $categoryInstanceTwo = CaseCategoryInstanceFabricator::fabricate();
+
+    // We call the method for getting the list.
+    $categoryInstances = (new CaseCategoryInstanceService())->getCaseCategoryInstances();
+
+    // And we verify the structure and content of the list received.
+    $this->assertArrayHasKey($categoryInstanceOne['id'], $categoryInstances);
+    $this->assertArrayHasKey($categoryInstanceTwo['id'], $categoryInstances);
+    $this->assertEquals(
+      $categoryInstanceOne['category_id'],
+      $categoryInstances[$categoryInstanceOne['id']]->category_id
+    );
+    $this->assertEquals(
+      $categoryInstanceOne['instance_id'],
+      $categoryInstances[$categoryInstanceOne['id']]->instance_id
+    );
+    $this->assertEquals(
+      $categoryInstanceTwo['category_id'],
+      $categoryInstances[$categoryInstanceTwo['id']]->category_id
+    );
+    $this->assertEquals(
+      $categoryInstanceTwo['instance_id'],
+      $categoryInstances[$categoryInstanceTwo['id']]->instance_id
+    );
```

Comment:
Can we also assert that `$categoryInstances` array only contains two records?

Reply:
As you can see [here](https://github.com/compucorp/uk.co.compucorp.civicase/pull/657/commits/6e359c8c0e7c363443ab1632103bb96283c370a6) and [here](https://github.com/compucorp/uk.co.compucorp.civicase/pull/657/commits/c7ee6d044dae774e40102e36d4f091224cbed39f), I added the assertion and then removed it, because it makes the test fail.

Before your review, when I first did this test, I asserted over the number of results, but I observed that I got one more than expected, based on some initially seeded data (I think that I modified the test database and now that error did not appear on my local, I will regenerate the DB after this). 

Now thinking a little deeper on the problem, I could get the base number of results (before calling the fabricator) and then verify that I have `base+2` results later, but I don't think it will be adding value to the test. What do you think?
I feel that adding two instances and verify that they are contained on the results, plus the later test filtering by instance type, provides good coverage. But I am open to suggestions.

Reply:
For the purpose of this test, we can add a method that cleans up case category instances in that table before this test runs, like a setup method (Maybe we just need it for this test but if we need it for all tests we can just add it in the setUp method). 
In that case, we are sure that there are no case category instance before we run the test and we can check that only two are created. It might also serve as a test to make sure that no more case category instance is created than expected.

Reply:
Sounds good. Done here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/657/commits/521300e71fa28319abbf247f38dcbb37e9ca6665
I added for this test and the other that executes the same method but with a filter parameter. The remainder method for the service does not require this cleaning, that's why I did not add it on the `setUp` call.